src/util/Xhr.js

```diff
@@ -69,39 +81,90 @@ class Xhr {
         sharedLinkPassword,
         responseInterceptor,
         requestInterceptor,
+        shouldRetry = true,
     }: Options = {}) {
         this.id = id;
         this.token = token;
         this.clientName = clientName;
         this.version = version;
         this.sharedLink = sharedLink;
         this.sharedLinkPassword = sharedLinkPassword;
-        this.responseInterceptor = responseInterceptor;
+        this.responseInterceptor = responseInterceptor || this.defaultResponseInterceptor;
         this.axios = axios.create();
         this.axiosSource = axios.CancelToken.source();
+        this.retryCount = 0;
+        this.shouldRetry = shouldRetry;
 
-        if (typeof responseInterceptor === 'function') {
-            // Called on any non 2xx response
-            this.axios.interceptors.response.use(responseInterceptor, this.errorInterceptor);
-        }
+        this.axios.interceptors.response.use(this.responseInterceptor, this.errorInterceptor);
 
         if (typeof requestInterceptor === 'function') {
             this.axios.interceptors.request.use(requestInterceptor);
         }
     }
 
+    /**
+     * Default response interceptor which just returns the response
+     *
+     * @param {Object} response - the axios response
+     * @return the response
+     */
+    defaultResponseInterceptor(response: $AxiosXHR<any>) {
+        return response;
+    }
+
+    /**
+     * Determines if a request should be retried
+     *
+     * @param {Object} error - Error object from axios
+     * @return {boolean} true if the request should be retried
+     */
+    shouldRetryRequest(error: $AxiosError<any>): boolean {
+        if (!this.shouldRetry || this.retryCount >= MAX_NUM_RETRIES) {
+            return false;
+        }
+
+        const { response, request } = error;
+        // Retry if there is a network error (e.g. ECONNRESET) or rate limited
+        const isNetworkError = request && !response;
+        const isRetryableStatusCode = isNetworkError || getProp(response, 'status') === HTTP_STATUS_CODE_RATE_LIMIT;
+        return isRetryableStatusCode;
+    }
+
+    /**
+     * Calculate the exponential backoff time with randomized jitter. Taken from box node SDK
+     *
+     * @param {number} numRetries Which retry number this one will be. Must be > 0
+     * @returns {number} The number of milliseconds after which to retry
+     */
+    getExponentialRetryTimeoutInMs(numRetries: number): number {
+        const minRandomization = 1 - RETRY_RANDOMIZATION_FACTOR;
+        const maxRandomization = 1 + RETRY_RANDOMIZATION_FACTOR;
+        const randomization = Math.random() * (maxRandomization - minRandomization) + minRandomization;
```

Comment:
Since `RETRY_RANDOMIZATION_FACTOR` is `0.5`, `maxRandomization - minRandomization` will always be `1`.
Simplify to `Math.random() + minRandomization`

Reply:
This is true, it was a copy + paste from node SDK. I did some more research and simplified the algorithm based off https://cloud.google.com/iot/docs/how-tos/exponential-backoff